src/github-graphql-base.js

```diff
@@ -0,0 +1,612 @@
+        'cursor:', cursor, 'page count:', pageCount);
+    }
+
+    let data, errors, hasPreviousPage = false, reachDateThreshold = false;
+    try {
+      ({ data, errors } = await this.client.query(
+        GITHUB_PR_INFO_QUERY,
+        {owner, name, perPage, before: cursor, orderBy}
+      ));
+    } catch (error) {
+      console.log('fetch pull request fails, owner:', owner, 'name:', name,
+        'cursor:', cursor, 'error:', error);
+    }
+    if (data && data.repository) {
+      // update rate limit
+      this._updateRateLimit(data.rateLimit);
+
+      this.pageCount++;
+      const { nodes, pageInfo } = data.repository.pullRequests;
+      this.cursor = pageInfo.startCursor;
+      hasPreviousPage = pageInfo.hasPreviousPage;
+      const result = await Promise.all(nodes.map(async node => {
+        if (getTime(node.updatedAt) < getTime(earliestDate)) {
+          reachDateThreshold = true;
+          return null;
+        }
+        let info = this._mapNodeToIssue(node);
+
+        let rawComments, comments;
+        // collect review comments
+        rawComments = node.reviews.nodes.map(node => node.comments.nodes);
+        rawComments = [].concat.apply([], rawComments);
+        // collect issue comments
+        rawComments = rawComments.concat(node.comments.nodes);
+
+        // fetch reactions only if there are reactions within that PR
+        // to reduce API hits.
+        // trick: a comment has received reaction(s) iff
+        // reactionGroup.createdAt attribute is not null
+        const hasReactions = rawComments.reduce((has, rawComment) => {
+          return has || rawComment.reactionGroups.reduce((flag, node) => {
+            return (flag || node.createdAt) ? true : false;
+          }, false);
+        }, false);
+
+        if (hasReactions) {
+          if (this.debug) {
+            console.log('pull request has reactions. repoOwner:', owner,
+              'repoName:', name, 'pullRequest number:', node.number);
+          }
+          const number = node.number;
+          const reviewsCount = Math.min(node.reviews.totalCount, 20);
+          // max number of discussions within a review
+          const discussionsPerReview = Math.min(
+            node.reviews.nodes.reduce(
+              (max, review) => Math.max(max, review.comments.totalCount), 0),
+            100);
+          const commentsCount = Math.min(node.comments.totalCount, 100);
+
+          const commentsWithReactions = await new GraphQLClientBase(this.token,
+            this.debug, this.sleepTime, 3)
+            .repo(this.repoOwner, this.repoName)
+            .reactions({pr_number: number,
+              reviews_count: reviewsCount,
+              discussions_per_review: discussionsPerReview,
+              comments_count: commentsCount})
+            .fetchOne();
+
+          // merge rawComments with commentsWithReactions
+          if (commentsWithReactions && commentsWithReactions.length) {
```

Comment:
Maybe check if `commentsWithReactions` is the same length as `rawComments` here because if not, you might have an error.